packages/core/nuxt-module/lib/plugins/i18n-cookies.js

```diff
@@ -69,20 +70,24 @@ const i18nCookiesPlugin = ({ $cookies, i18n, app, redirect }) => {
   if (missingFields.length) {
     throw new Error(`Following fields are missing in the i18n configuration: ${missingFields.join(', ')}`);
   }
-
-  if (cookieLocale !== settings.locale) {
+  if (cookieLocale !== settings.locale && !!i18nOptions.autoChangeCookie?.locale || !$cookies.get(cookieNames.locale)) {
```

Comment:
There's no point in checking if cookie exists as its done by first condition: `cookieLocale !== settings.locale`